src/node.rs

```diff
@@ -646,6 +654,31 @@ impl<S: std::fmt::Debug + ForkSource> InMemoryNodeInner<S> {
     pub fn stop_impersonating_account(&mut self, address: Address) -> bool {
         self.impersonated_accounts.remove(&address)
     }
+
+    /// Archives the current state for later queries.
+    pub fn archive_state(&mut self) -> Result<(), String> {
+        if self.previous_states.len() > MAX_PREVIOUS_STATES as usize {
+            if let Some(entry) = self.previous_states.shift_remove_index(0) {
+                log::debug!("removing archived state for previous block {:#x}", entry.0);
+            }
+        }
+        println!(
```

Comment:
Nit: use logger please

Reply:
That wasn't supposed to be there, but I guess it's nice to have in debug log.